lib/parse.js

```diff
@@ -1,5 +1,77 @@
 const util = require('./util')
 
+const ParseState = {
+    start: 0,
+    beforePropertyName: 1,
+    afterPropertyName: 2,
+    beforePropertyValue: 3,
+    afterPropertyValue: 4,
+    beforeArrayValue: 5,
+    afterArrayValue: 6,
+    end: 7,
+}
+
+const LexState = {
+    default: 8,
+    comment: 9,
+    multiLineComment: 10,
+    multiLineCommentAsterisk: 11,
+    singleLineComment: 12,
+    value: 13,
+    identifierNameStartEscape: 14,
+    identifierName: 15,
+    identifierNameEscape: 16,
+    sign: 17,
+    zero: 18,
+    decimalInteger: 19,
+    decimalPointLeading: 20,
+    decimalPoint: 21,
+    decimalFraction: 22,
+    decimalExponent: 23,
+    decimalExponentSign: 24,
+    decimalExponentInteger: 25,
+    hexadecimal: 26,
+    hexadecimalInteger: 27,
+    string: 28,
+    start: ParseState.start,
+    beforePropertyName: ParseState.beforePropertyName,
+    afterPropertyName: ParseState.afterPropertyName,
+    beforePropertyValue: ParseState.beforePropertyValue,
+    afterPropertyValue: ParseState.afterPropertyValue,
+    beforeArrayValue: ParseState.beforeArrayValue,
+    afterArrayValue: ParseState.afterArrayValue,
+    end: ParseState.end,
+}
+
+const Token = {
+    eof: 0,
+    punctuator: 1,
+    null: 2,
+    boolean: 3,
+    numeric: 4,
+    identifier: 5,
+    numeric: 6,
+    string: 7,
+    punctuator: 8,
+}
+
+const tokenRef = {
+    type: -1,
+    value: '',
+    line: -1,
+    column: -1,
+}
+
+const FIRST_DOUBLE_LENGTH_CODE = 1 << 16
+const getCodePointLength = c => c >= FIRST_DOUBLE_LENGTH_CODE ? 2 : 1
+
+const LITERAL_ULL     = [117 /* 'u' */, 108 /* 'l' */, 108 /* 'l' */]
+const LITERAL_RUE     = [114 /* 'r' */, 117 /* 'u' */, 101 /* 'e' */]
+const LITERAL_ALSE    = [97 /* 'a' */, 108 /* 'l' */, 115 /* 's' */, 101 /* 'e' */]
+const LITERAL_NFINITY = [110 /* 'n' */, 102 /* 'f' */, 105 /* 'i' */, 110 /* 'n' */, 105 /* 'i' */, 116 /* 't' */, 121 /* 'y' */]
+const LITERAL_AN      = [97 /* 'a' */, 78 /* 'N' */]
```

Comment:
`npm run lint` needs to be run before merging, which will condense the spacing here.

Let's also convert the decimal numbers that represent code points to 4-digit hexadecimal throughout the code base (i.e. `0x0075` instead of `117`). This will keep some of the other comments in line. 

That might be a bit of a pain to ask, but I know the hex values for characters better than the decimal ones. I can do that if you don't have the time or desire. Just let me know.

Reply:
No worries, this easy: `newCode=code.replaceAll(/(\d+) \/\*/g, (m, d) => '0x' + parseInt(d).toString(16) + ' /*')`